app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -37,7 +37,7 @@ def self.phishing_resistant?(auth_method)
     # rubocop:disable Rails/LexicallyScopedActionFilter
     before_action :authenticate_user
     before_action :require_current_password, if: :current_password_required?
-    before_action :check_already_authenticated
+    before_action :check_already_authenticated, unless: :in_reauthn_flow?
```

Comment:
Basically this method during verification of your authentication method somehow causes the user to bypass the verification and takes you back to the `after_sign_in_path`. We need a way to ensure that when a user is going through reauthn process they have to complete verification.